lib/ancestry/optimizers/pg.rb

```diff
@@ -0,0 +1,27 @@
+module Ancestry
+  module Optimizers
+    module Pg
```

Comment:
Right now, Rails 6 supports multiple databases and ActiveRecord has a way to establish new db connection. If there is a use case for switching between mysql and pg, then this approach will breaks. I'm thinking about changing it into class, and then we use a factory method to get the right optimizer for pg/mysql/... at runtime. It means `InstanceMethods#update_descendants_with_new_ancestry` change to:

```
def update_descendants_with_new_ancestry
   if optimizer
     optimizer.update_descendants_with_new_ancestry
  else
     # origin methods
  end
end
```

But I feel like it is still not good,  it is harder to maintain. Do you have any suggestions?

Reply:
I think the database type will probably be hard coded at design time, and definitely at deploy time.
It will not change at runtime.

But I do see pointing to serverless and a relational at the same time.
Find it doubtful they would point to mysql and postgres at the same time.